apps/backend/app/core/database.py

```
@@ -117,16 +212,81 @@ def get_sync_db_session() -> Generator[Session, None, None]:
+    try:
+        async with async_engine.connect() as conn:
+            await conn.execute(text("SELECT 1"))
+        return True
+    except Exception as e:
+        logger.error(f"Database connection check failed: {e}")
+        return False
+
+
+async def get_database_stats() -> dict:
+    """Get database connection pool statistics."""
+    pool = async_engine.pool
```

Ruff (0.11.9)

Redefinition of unused `pool` from line 9

(F811)